webapp/utils/markdown.jsx

```diff
@@ -103,7 +103,7 @@ class MattermostMarkdownRenderer extends marked.Renderer {
             output += ' title="' + title + '"';
         }
 
-        output += '>' + text + '</a>';
+        output += '>' + text.replace(/\//g, '/<wbr />') + '</a>';
```

Comment:
Thanks for making that change, but I think doFormatText is still the correct place for it. At this point, `text` has actually ran through doFormatText (it gets passed to MattermostMarkdownRenderer.text before getting passed to MattermostMarkdownRenderer.link), but it may have HTML in it from that. As it is, links of the form `[:smile:](www.google.ca)` break with your code